### ISSUE TITLE
Construction yields items in their default containers

### ISSUE THREAD

**Issue by Karol1223** [CleverRaven/Cataclysm-DDA#66853]
### Describe the bug

Construction recipes that yield items do so in the items' default container if it has one defined

### Attach save file

[Temstz-trimmed.tar.gz](https://github.com/CleverRaven/Cataclysm-DDA/files/12030683/Temstz-trimmed.tar.gz)

### Steps to reproduce

1. When you load the save you'll already be next to a taped window
2. Just enter the construction menu and remove tape from window
3. Stare in awe as the gods manifest the tape in a cardboard roll willed into existence specifically for you

### Expected behavior

Maybe no roll?????? Default containers shouldn't be manifested into reality like this.

### Versions and configuration

- OS: Windows
    - OS Version: 10.0.19045.3086 (22H2)
- Game Version: 2cd9d43 [64-bit]
- Graphics Version: Tiles
- Game Language: English [en]
- Mods loaded: [
    Dark Days Ahead [dda],
    Disable NPC Needs [no_npc_food],
    Portal Storms Ignore NPCs [personal_portal_storms],
    Slowdown Fungal Growth [no_fungal_growth],
    Bionic Professions [package_bionic_professions],
    Standard combat test suite [standard_combat_test]
]

### Additional context

For anyone loading the attached save - don't mind the dead children

**Comment by RenechCDDA** [comment]
Not really a bug? This is intended behavior.

**Comment by Karol1223** [comment]
> Not really a bug? This is intended behavior.

...in what way? Tearing off duct tape should spawn you a roll? Hypothetically, scooping up sand should spawn you a bag to hold this sand? This makes no sense in any way shape or form.

**Comment by RenechCDDA** [comment]
> Tearing off duct tape should spawn you a roll?

No, of course not. But you told the game that the duct tape should always be in this container. So it is.

**Comment by RenechCDDA** [comment]
See, e.g. #62148

Default container do be default

**Comment by Karol1223** [comment]
> > Tearing off duct tape should spawn you a roll?
> 
> No, of course not. But you told the game that the duct tape should always be in this container. So it is.

I have a hard time agreeing with this train of thought. If crafting duct tape doesn't spawn it in its container, then neither it should upon disassembly or deconstruction. Disassembly we don't have to worry about because it has ``NO_SALVAGE``, but regardless both of those are similar cases where the game should UNDER NO CIRCUMSTANCES spawn containers out of thin air. Apologies but I don't consider this reasoning anything even remotely reasonable.

**Comment by RenechCDDA** [comment]
So, just like I mentioned for byproducts groups, the way to resolve this 'bug' is to actually define the item without a container:

```
    "deconstruct": {
      "ter_set": "t_window_empty",
      "items": [ { "item": "glass_sheet", "count": 1 }, { "item": "duct_tape", "charges": [ 15, 25 ], "container-item": "null" } ]
    },
```


![image](https://github.com/CleverRaven/Cataclysm-DDA/assets/84619419/0a1e8e5a-6137-4d7d-971c-a4de95d4fe73)

**Comment by Karol1223** [comment]
> So, just like I mentioned for byproducts groups, the way to resolve this 'bug' is to actually define the item without a container:
> 
> ```
>     "deconstruct": {
>       "ter_set": "t_window_empty",
>       "items": [ { "item": "glass_sheet", "count": 1 }, { "item": "duct_tape", "charges": [ 15, 25 ], "container-item": "null" } ]
>     },
> ```
> 
> 
> ![image](https://github.com/CleverRaven/Cataclysm-DDA/assets/84619419/0a1e8e5a-6137-4d7d-971c-a4de95d4fe73)
> 

If I simply wanted to resolve the issue here I'd just remove the duct tape (I do believe that removing it from a window would not yield usable duct tape). I posted this because I fully believe that construction should not take default container as it is unreasonable. There are edge cases where it coild be useful, such as pallets of cement powder if cement powder had default containers, but then a hypothetical "pile of cement powder" furniture would suffer the confusion instead. If the people with the power to close this deem this as an unreasonable bug report so be it but I will not close it myself as I am of firm belief this does not behave as it should.

**Comment by RenechCDDA** [comment]
Everything takes default container. That's why it is default. (Yes, I agree that makes it very hard to have use cases for it, but **making it more inconsistent** helps nobody.)

Recipes not spawning default container is a byproduct(no pun intended) of their special casing.

**Comment by Karol1223** [comment]
> making it more inconsistent

I thoroughly disagree because in its current state for everything that's not a liquid it is literally safer to manually define containers in case it turns out it falls apart behind the corner. But I do think we've both said all that's to be said. Default containers leave a lot to be desired and I just want this to stay up for the month until stalebot comes around to at least be a warning for others that no, they're not as comfortable as they look. I was hesitant to use them and I do regret it by now.

**Comment by GuardianDll** [comment]
I personally believe that default container field should be gone, and never be used no matter how convenient it is, simply because it would be entirely item group-driven thing, insteand of "50% in the loot group definition, 50% on the item defintion"

**Comment by Procyonae** [comment]
Removing it entirely does seem like a much better option, I can't think of any behaviour we can't replicate without it off the top of my head, the only slightly annoying thing is you can't specify a container group for a whole item group from my testing (you can to each item in the group though) but that should be a fairly minor fix
